impy/common.py

```diff
@@ -18,6 +18,28 @@
 from impy.util import info
 
 
+class EventData:
+    """Data structure to keep views of filtered data"""
+    def __init__(self, npart, p_ids, status, charge,
+                 px, py, pz, en, m, vx, vy, vz, vt,
+                 pem_arr, vt_arr):
+        self.npart = npart
+        self.p_ids = p_ids
+        self.status = status
+        self.charge = charge
+        self.px = px
+        self.py = py
+        self.pz = pz
+        self.en = en
+        self.m = m
+        self.vx = vx
+        self.vy = vy
+        self.vz = vz
+        self.vt = vt
```

Comment:
Perhaps these arrays should be made read-only, so that users cannot accidentally override the values?

Reply:
These are views into the FORTRAN common block, right? What is the memory layout of this block? Is it
px1 px2 px3 px4 ... py1 py2 py3 ... etc or is it px1 py1 ... px2 py2 ... ? If it is the latter, the view could be a single record array. This would make the masking/filtering easier.

Reply:
class EventData was needed only for the cache of views. If `px`, `py`, `pz`, ... are properties then EventData will be removed

Reply:
Yes, that's what I had in mind. If we know the layout of the underlying memory block, we can apply the mask very efficiently only once.

Reply:
I would like to experiment a bit with filtering in a different PR. I think for this PR, the code is fine.